arch/arc/core/isr_wrapper.S

```diff
@@ -230,47 +230,38 @@ From RIRQ:
  */
 
 SECTION_FUNC(TEXT, _isr_wrapper)
+#if CONFIG_ARC_FIRQ
+#if CONFIG_RGF_NUM_BANKS == 1
+	st r0,[saved_r0]
+#endif
+
 #ifdef CONFIG_EXECUTION_BENCHMARKING
+#if CONFIG_RGF_NUM_BANKS !=1
```

Comment:
Why switching banks here?
And more important: shouldn't the current bank (depending on entering via FIRQ or RIRQ) be saved instead of always restored to bank 1 later (line 260).

Reply:
1. FIRQ's priroity is highest,  impossible to have  nest firq.  So the interruptted reg bank must be bank 0. 
2.  As some pop and push are required, we need to switch back to the interruptted bank to get the right sp.
3. switch back to bank1 means to recover the context of firq,  do not impact the following  interrupt handling

Reply:
and there is a better place to call `read_timer_start_of_isr`, which is simpler,    It's at the beiging of  _isr_demux.,  but the side effect is that the perf number may be not accurate.

Reply:
@ruudw, i add another simpler and cleaner commit with a result of not accurate latency number